homeassistant/components/keenetic_ndms2/device_tracker.py

```diff
@@ -1,102 +1,161 @@
+        self.router = router
+        self._last_seen = (
+            dt_util.utcnow() if device.mac in router.last_devices else None
         )
 
-        self.success_init = self._update_info()
-        _LOGGER.info("Scanner initialized")
-
-    def scan_devices(self):
-        """Scan for new devices and return a list with found device IDs."""
-        self._update_info()
-
-        return [device.mac for device in self.last_results]
-
-    def get_device_name(self, device):
-        """Return the name of the given device or None if we don't know."""
-        name = next(
-            (result.name for result in self.last_results if result.mac == device), None
+    @property
+    def should_poll(self) -> bool:
+        """Return False since entity pushes its state to HA."""
+        return False
+
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the network."""
+        return (
+            self._last_seen
+            and (dt_util.utcnow() - self._last_seen)
+            < self.router.consider_home_interval
         )
-        return name
 
-    def get_extra_attributes(self, device):
-        """Return the IP of the given device."""
-        attributes = next(
-            ({"ip": result.ip} for result in self.last_results if result.mac == device),
-            {},
+    @property
+    def source_type(self):
+        """Return the source type of the client."""
+        return SOURCE_TYPE_ROUTER
+
+    @property
+    def name(self) -> str:
+        """Return the name of the client."""
+        return self.device.name
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this device."""
+        return self.device.mac
+
+    @property
+    def available(self) -> bool:
+        """Return if controller is available."""
+        return self.router.available
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        if self.is_connected:
+            return {"ip": self.device.ip}
+        return None
+
+    @property
+    def device_info(self):
+        """Return a client description for device registry."""
+        info = {
+            "connections": {(CONNECTION_NETWORK_MAC, self.device.mac)},
+            "identifiers": {(DOMAIN, self.device.mac)},
+            "name": self.name,
+            "via_device": (DOMAIN, self.router.host),
+        }
+        return info
+
+    async def async_added_to_hass(self):
+        """Client entity created."""
+        _LOGGER.debug("New network device tracker %s (%s)", self.name, self.unique_id)
+
+        @callback
+        def update_device():
+            _LOGGER.debug(
+                "Updating Keenetic tracked device %s (%s)",
+                self.entity_id,
+                self.unique_id,
+            )
+            new_device = self.router.last_devices.get(self.device.mac)
+            if new_device:
+                self.device = new_device
```

Comment:
We do we need to set the device again in the update? The mac address will be the same and the name shouldn't change since we have a unique id set.

Reply:
Device IP and name can change on the router side: IP can be dynamically allocated while name can be changed by the router admin

Reply:
Ok. FYI the name change will not be applied dynamically if the user has set a name since we set a unique id. Only if the user updates the entity in the entity registry will the default name be changed there.

Reply:
OK. I've checked the case with the name change on the router side (in case user does not set a name):
* name of the device_tracker entity changed in UI
* the name of device instance stays the same

Then after HASS restart the name of the device changed also

When I customize the name in UI changes from the router are ignored.